### PR TITLE
Compare function

### PR DESCRIPTION
**Pull request by Szmajasz**

### CODE REVIEW

File: R/compare_models.R [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+#' lightgbm_explained <- make_lightgbm(apartments, "m2.price", "regression")
+#' ranger_explained <- make_ranger(apartments, "m2.price", "regression")
+#' 
+#' models <- list(lightgbm_explained, ranger_explained)
+#' 
+#' best_model <- compare_models(models, apartments, "m2.price", "rmse")
+#'}
+##
+
+compare_models <- function(models, data_test, target, metric = NULL){
```
Comment (kozaka93) [inline R/compare_models.R:33]: In this line we can use "..." as an agument. Then we can call the function as compare_models(model1, model2, model3, other parameters).

File: R/compare_models.R [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    stop("Models should be passed as a list")
+  }
+  
+  if (length(models) == 0){
+    stop("List of models is empty.")
+  }
+  
+  # Variable for checking models type 
+  if (!any(class(models[[1]]) %in% c("forester_model", "explainer"))){
+    stop("Wrong class of model. Models should have forester_model or explainer class.")
```
Comment (kozaka93) [inline R/compare_models.R:46]: Maybe "Wrong class of model. Model should be build with forester function." or something like that.

File: R/compare_models.R [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    col_names <- c("model", metric, non_imprtant_metrics)
+    results <- results[col_names]
+  }
+  
+  # Sorting results based on chosen metric
+  final <- results[order(results[[metric]],
+                         decreasing = as.logical(TRUE + negative)),]
+  
+  # Printing table
+  print(knitr::kable(final, row.names = F))
```
Comment (kozaka93) [inline R/compare_models.R:113]: Print table in other function.

File: R/compare_models.R [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  
+  return(best_model)
```
Comment (kozaka93) [inline R/compare_models.R:115]: Best_model or final or both - what we return?

File: R/forester.R [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      # Unquote quotation marks
+      processed_label <- catboost_exp$y
+      uniq_label <- unique(processed_label)
+      data_test[[target]] <- ifelse(data_test[[target]] == data_train[[target]][1],
+                                 uniq_label[1],
+                                 uniq_label[2])
+  }
+  
+  best_model <- compare_models(models, data_test, target, metric)
+
```
Comment (kozaka93) [inline R/forester.R:86]: Add return() and add class.

Comment (Szmajasz) [reply R/forester.R:86]: I've added return(). Compare function returns "forester_model" object so best model has already a class.

**Pull request merged**